### PR TITLE
Definition of Done pull request template

### PR DESCRIPTION
**Pull request by nikipuk**
## Related Issues

- Closes #
- Fixes #

## Definition of Done

- [ ] User Story
  - [ ] All Acceptance Criteria met
- [ ] Code
  - [ ] No more code needed
  - [ ] No known bugs
- [ ] Clean Code
  - [ ] Comments have been created for new code
  - [ ] ESLinter found no issues (editor & executor only)
  - [ ] No unavoidable Code Smells
- [ ] Testing
  - [ ] Existing Tests pass
  - [ ] New Unit Tests created
  - [ ] New Unit Tests pass
- [ ] SonarCloud Quality Gate passed

## Definition of more than Done (optional)

- [ ] Testing
  - [ ] New Integration Tests

### CODE REVIEW

File: docs/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+## Is this UserStory done?
+
+You might be confused about what it really means for a User Story to be 'done'.
```
Comment (MrF3lix) [inline docs/pull_request_template.md:3]: This this belong in every pull request? :D

Anyway, I would also add a section to remind people to link their issues. 

```
## Related Issues

- Closes #
- Fixes #
```

File: docs/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Well you're in luck! We at the **DoD - Department of Done** have comprised a helpful checklist to see if your code meets our one true definition of done.
+
+The User Story is done if **and only if**:
+
+- [ ] User Story
+- - [ ] All Acceptance Criteria met
```
Comment (MrF3lix) [inline docs/pull_request_template.md:10]: You don't need two `-` in markdown. Just indent them using spaces.

File: docs/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+- - [ ] No more code needed
+- - [ ] No known bugs
+- [ ] Clean Code
+- - [ ] Comments have been created for new code
+- - [ ] ESLinter found no issues
+- - [ ] No unavoidable Code Smells
+- [ ] Testing
+- - [ ] Existing Tests pass
+- - [ ] New Unit Tests created
+- - [ ] New System Tests (if possible)
```
Comment (MrF3lix) [inline docs/pull_request_template.md:21]: I would create a separate check list that is optional. And also, what are system tests?

File: docs/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+The User Story is done if **and only if**:
+
+- [ ] User Story
+- - [ ] All Acceptance Criteria met
+- [ ] Code
+- - [ ] No more code needed
+- - [ ] No known bugs
+- [ ] Clean Code
+- - [ ] Comments have been created for new code
+- - [ ] ESLinter found no issues
```
Comment (MrF3lix) [inline docs/pull_request_template.md:16]: API uses dotnet and doesn't have eslint configured. How are we dealing with checkboxes that are only for editor/executor? or only for the api?

**Review by MrF3lix — CHANGES_REQUESTED** [review]
Maybe also mention somewhere that the sonar quality gate should pass? I mean a lot of the checkboxes are covered automatically in the defined quality gate.

File: docs/pull_request_template.md (hunk 4 above)
Comment (DuplosFidibuss) [reply docs/pull_request_template.md:16]: I think one could leave these checkboxes where they are and just add editor/executor in brackets.

File: docs/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+## Is this UserStory done?
+
+You might be confused about what it really means for a User Story to be 'done'.
+
+Well you're in luck! We at the **DoD - Department of Done** have comprised a helpful checklist to see if your code meets our one true definition of done.
+
+The User Story is done if **and only if**:
```
Comment (DuplosFidibuss) [inline docs/pull_request_template.md:7]: Fancy, but according to me not necessary in the template. Add the section "Related issues" instead (as mentionned by saarofel).

**Review by DuplosFidibuss — CHANGES_REQUESTED** [review]
Looks good overall, please have a look at the comments for changes.

File: docs/pull_request_template.md (hunk 5 above)
Comment (nikipuk) [reply docs/pull_request_template.md:7]: yea, I was going to remove it after the review, I was just in the mood for some dumb stuff

**Review by DuplosFidibuss — APPROVED** [review]

**Review by MrF3lix — APPROVED** [review]

**Pull request merged**